### ISSUE TITLE
[BUG] NumPy 1.19 compatibility

### ISSUE THREAD

**Issue by mnicely** [rapidsai/cusignal#166]
It looks like we need to wait on CuPy for finish proving out NumPy 1.19 compatibility with CuPy.

This should follow the following CuPy PRs.
https://github.com/cupy/cupy/issues/3627
https://github.com/cupy/cupy/issues/3630

Currently we are seeing the following error on sparatic functions. Below is from pytest-benchmark

```bash
a = (array([1.        , 0.99980783, 0.99923141, ..., 0.        , 0.        ,
       0.        ]), array([ 0.        ,  0.0...0.        ,  0.        ]), array([1.        , 0.99999168, 0.99996673, ..., 0.        , 0.        ,
       0.        ]))
dtype = None, order = 'C'

    @set_module('numpy')
    def asarray(a, dtype=None, order=None):
        """Convert the input to an array.
    
        Parameters
        ----------
        a : array_like
            Input data, in any form that can be converted to an array.  This
            includes lists, lists of tuples, tuples, tuples of tuples, tuples
            of lists and ndarrays.
        dtype : data-type, optional
            By default, the data-type is inferred from the input data.
        order : {'C', 'F'}, optional
            Whether to use row-major (C-style) or
            column-major (Fortran-style) memory representation.
            Defaults to 'C'.
    
        Returns
        -------
        out : ndarray
            Array interpretation of `a`.  No copy is performed if the input
            is already an ndarray with matching dtype and order.  If `a` is a
            subclass of ndarray, a base class ndarray is returned.
    
        See Also
        --------
        asanyarray : Similar function which passes through subclasses.
        ascontiguousarray : Convert input to a contiguous array.
        asfarray : Convert input to a floating point ndarray.
        asfortranarray : Convert input to an ndarray with column-major
                         memory order.
        asarray_chkfinite : Similar function which checks input for NaNs and Infs.
        fromiter : Create an array from an iterator.
        fromfunction : Construct an array by executing a function on grid
                       positions.
    
        Examples
        --------
        Convert a list into an array:
    
        >>> a = [1, 2]
        >>> np.asarray(a)
        array([1, 2])
    
        Existing arrays are not copied:
    
        >>> a = np.array([1, 2])
        >>> np.asarray(a) is a
        True
    
        If `dtype` is set, array is copied only if dtype does not match:
    
        >>> a = np.array([1, 2], dtype=np.float32)
        >>> np.asarray(a, dtype=np.float32) is a
        True
        >>> np.asarray(a, dtype=np.float64) is a
        False
    
        Contrary to `asanyarray`, ndarray subclasses are not passed through:
    
        >>> issubclass(np.recarray, np.ndarray)
        True
        >>> a = np.array([(1.0, 2), (3.0, 4)], dtype='f4,i4').view(np.recarray)
        >>> np.asarray(a) is a
        False
        >>> np.asanyarray(a) is a
        True
    
        """
>       return array(a, dtype, copy=False, order=order)
E       ValueError: object __array__ method not producing an array

../../../anaconda3/envs/cusignal/lib/python3.8/site-packages/numpy/core/_asarray.py:83: ValueError
```

**Comment by kkraus14** [comment]
@mnicely that error is due to something trying to call `np.asarray(...)` against a cupy array which has been decided they want to throw instead of implicitly doing a D2H copy. Do you have a test failure or stacktrace as to where this is occurring?

**Comment by mnicely** [comment]
Currently, I'm seeing it occur when we run pytest. On `cp.asnumpy` 
What's confusing is that we use `cp.asnumpy` on each test. I don't know why it only errors on a handful. :thinking:

**Comment by mnicely** [comment]
The issue was in our pytest and never received errors before. This bug will be closed with #150
